pkg/cloud/externalconn/impl_registry.go

```diff
@@ -138,11 +138,71 @@ func ExternalConnectionFromURI(
 		return nil, errors.Newf("no parseAndValidateFn found for external connection provider %s", externalConnectionURI.Scheme)
 	}
 
-	return parseAndValidateFn.parseAndValidateURI(ctx, execCfg, user, externalConnectionURI, defaultValidation)
+	return parseAndValidateFn.parseAndValidateURI(ctx, env, externalConnectionURI, defaultValidation)
 }
 
-// ValidationFn is the type taken by Register*Validation. Validation functions can ping the external connection.
-type ValidationFn func(ctx context.Context, execCfg interface{}, user username.SQLUsername, uri string) error
+type ExternalConnEnv struct {
+	// Settings refers to the cluster settings that apply to the BackupKMSEnv.
+	Settings *cluster.Settings
+	// Conf represents the ExternalIODirConfig that applies to the BackupKMSEnv.
+	Conf *base.ExternalIODirConfig
+	// DB is the database handle that applies to the BackupKMSEnv.
+	Isqldb isql.DB
+	// Username is the user that applies to the BackupKMSEnv.
+	Username username.SQLUsername
+
+	ExternalStorageFromURIFactory cloud.ExternalStorageFromURIFactory
+
+	SkipCheckingExternalStorageConnection bool
+	SkipCheckingKMSConnection             bool
+	// ServerCfg is used to validate the external connection sink URI.
+	//// It implements *execinfra.ServerConfig.
+	ServerCfg interface{}
+	//ServerCfg *execinfra.ServerConfig
+}
+
+var _ cloud.KMSEnv = &ExternalConnEnv{}
+
+func (s *ExternalConnEnv) ClusterSettings() *cluster.Settings {
+	return s.Settings
+}
+
+func (s *ExternalConnEnv) KMSConfig() *base.ExternalIODirConfig {
+	return s.Conf
+}
+
+func (s *ExternalConnEnv) DBHandle() isql.DB {
+	return s.Isqldb
+}
+
+func (s *ExternalConnEnv) User() username.SQLUsername {
+	return s.Username
+}
+
+func MakeExternalConnEnv(
+	settings *cluster.Settings,
+	conf *base.ExternalIODirConfig,
+	isqlDB isql.DB,
+	username username.SQLUsername,
+	externalStorageFromURIFactory cloud.ExternalStorageFromURIFactory,
+	skipCheckingExternalStorageConnection bool,
+	skipCheckingKMSConnection bool,
+	serverCfg *execinfra.ServerConfig,
```

Comment:
should this be interface{} otherwise we're still pulling in `execinfra`?

Reply:
Done.